server/src/transport/http.rs

```diff
@@ -50,6 +50,20 @@ pub(crate) async fn reject_connection(socket: tokio::net::TcpStream) {
 	}
 }
 
+/// The `Authority` can be sent by the client in the `Host header` or in the `URI`
+/// such that we must check both.
+pub(crate) fn fetch_authority(request: &hyper::Request<hyper::Body>) -> Option<&str> {
+	let host_header = http_helpers::read_header_value(request.headers(), hyper::header::HOST);
+	let uri = request.uri().authority();
```

Comment:
This is fix and mainly regards HTTP2 but could be some edge-case as well.

Thus, if one would enable a filter for `port X` and the request is targets port `X` but it's not read and would regarded as the default port if just the `host` is read i.e, not the port